public/src/components/Home/ArchivedSubredditsComponent.jsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import ArchivedSubredditItem from './ArchivedSubredditItem';
+
+const ScrollArea = require('react-scrollbar').default;
+
+export default class ArchivedSubredditsComponent extends React.Component {
+    getErrorMessages() {
+        const { errorMessages } = this.props;
+        return errorMessages.map((message, index) => (
+            <div className="errorMessage" key={index}>{message}</div>
+        ));
+    }
+
+    getArchivedSubredditsList() {
+        const {
+            subreddits,
+            archiveFormValue,
+            storeSubredditToArchive,
+            removePosts,
+        } = this.props;
+
+        const subredditList = [...subreddits].map((subreddit, key) => {
+            const formValue = archiveFormValue.toLowerCase().trim();
+            const subredditTitle = subreddit.display_name.toLowerCase().trim();
+
+            if (subredditTitle.search(formValue) > -1 && subreddit.isArchived) {
+                return (
+                    <ArchivedSubredditItem key={key}
+                                           storeSubredditToArchive={storeSubredditToArchive}
+                                           removePosts={removePosts}
+                                           subreddit={subreddit}
+                    />
+                );
+            }
+            return null;
+        });
+
+        if (subredditList.some(subreddit => subreddit)) {
+            return (
+                <div className="subredditList-inner_container">
+                    <ScrollArea speed={0.8}
+                                horizontal={false}
+                    >
+                        {subredditList}
+                    </ScrollArea>
+                </div>
+            );
+        }
+        return <span className="subredditList-inner_container empty">...</span>;
+    }
+
+    render() {
+        return (
+            <div className="home-sidebar__archive-form__list">
+                <span className="home-sidebar__archive-form__title">Archived subreddits:</span>
+                {this.getArchivedSubredditsList.call(this)}
+                {this.getErrorMessages.call(this)}
```

Comment:
In some files you use `.call(this)` and in some files you bind functions in constructor, maybe just bind functions for consistency in every file?

Reply:
I'm using .call(this) when i want to get computer properties, like styles, list of items, etc. 
I'm using .bind, when i have some method to handle event, or some method i want to pass as prop to child component.
I think, it's conveniently, don't you think so?